libp2p/protocols/mix/exit_connection.nim

```diff
@@ -0,0 +1,68 @@
+import hashes, chronos, chronicles
+import ../../stream/connection
+from fragmentation import DataSize
+
+type MixExitConnection* = ref object of Connection
+  message: seq[byte]
+  response: seq[byte]
+
+method readOnce*(
+    self: MixExitConnection, pbytes: pointer, nbytes: int
+): Future[int] {.async: (raises: [CancelledError, LPStreamError], raw: true), public.} =
+  let responseFut = newFuture[int]()
+  if self.message.len == 0:
+    responseFut.complete(0) # Nothing else to read.
+    return responseFut
+  if self.message.len < nbytes:
+    responseFut.fail(
+      newException(
+        LPStreamError, "Not enough data in to read exactly " & $nbytes & " bytes."
+      )
+    )
+    return responseFut
+  copyMem(pbytes, addr self.message[0], nbytes)
+  self.message = self.message[nbytes ..^ 1]
+  responseFut.complete(nbytes)
+  responseFut
+
+method write*(
+    self: MixExitConnection, msg: seq[byte]
+): Future[void] {.async: (raises: [CancelledError, LPStreamError], raw: true), public.} =
+  if msg.len() > DataSize:
+    let fut = newFuture[void]()
+    fut.fail(
+      newException(LPStreamError, "exceeds max msg size of " & $DataSize & " bytes")
+    )
+    return fut
+
+  self.response.add(msg)
+  let fut = newFuture[void]()
+  fut.complete()
+  return fut
```

Comment:
why you `raw`? code would be more simpler if it is not, here and other places in mix.

Reply:
I just tried without raw and it works. I did try on diff PR and had to use raw there. Probably did something dumb back then... Will update the PR